src/core/mantle/data/gtfsReader.ts

```diff
@@ -0,0 +1,454 @@
+  };
+
+  protected readField(tag: number, obj = {} as VehiclePosition, pbf: Pbf): void {
+    if (tag === 1) obj.trip = new TripDescriptorReader().read(pbf, pbf.readVarint() + pbf.pos);
+    else if (tag === 8)
+      obj.vehicle = new VehicleDescriptorReader().read(pbf, pbf.readVarint() + pbf.pos);
+    else if (tag === 2) obj.position = new PositionReader().read(pbf, pbf.readVarint() + pbf.pos);
+    else if (tag === 3) obj.current_stop_sequence = pbf.readVarint();
+    else if (tag === 7) obj.stop_id = pbf.readString();
+    else if (tag === 4) obj.current_status = pbf.readVarint();
+    else if (tag === 5) obj.timestamp = pbf.readVarint();
+    else if (tag === 6) obj.congestion_level = pbf.readVarint();
+    else if (tag === 9) obj.occupancy_status = pbf.readVarint();
+  }
+}
+
+export enum VehicleStopStatus {
+  INCOMING_AT,
+  STOPPED_AT,
+  IN_TRANSIT_TO,
+}
+
+export enum CongestionLevel {
+  UNKNOWN_CONGESTION_LEVEL,
+  RUNNING_SMOOTHLY,
+  STOP_AND_GO,
+  CONGESTION,
+  SEVERE_CONGESTION,
+}
+
+export enum OccupancyStatus {
+  EMPTY,
+  MANY_SEATS_AVAILABLE,
+  FEW_SEATS_AVAILABLE,
+  STANDING_ROOM_ONLY,
+  CRUSHED_STANDING_ROOM_ONLY,
+  FULL,
+  NOT_ACCEPTING_PASSENGERS,
+}
+
+export class AlertReader extends ProtobufMessageReader<Alert> {
+  readonly defaultMessage: Alert = {
+    active_period: [],
+    informed_entity: [],
+    cause: AlertCause.CONSTRUCTION,
+    effect: AlertEffect.STOP_MOVED,
+    url: null,
+    header_text: null,
+    description_text: null,
+  };
+
+  protected readField(tag: number, obj = {} as Alert, pbf: Pbf): void {
+    if (tag === 1)
+      obj.active_period?.push(new TimeRangeReader().read(pbf, pbf.readVarint() + pbf.pos));
+    else if (tag === 5)
+      obj.informed_entity?.push(new EntitySelectorReader().read(pbf, pbf.readVarint() + pbf.pos));
+    else if (tag === 6) obj.cause = pbf.readVarint();
+    else if (tag === 7) obj.effect = pbf.readVarint();
+    else if (tag === 8)
+      obj.url = new TranslatedStringReader().read(pbf, pbf.readVarint() + pbf.pos);
+    else if (tag === 10)
+      obj.header_text = new TranslatedStringReader().read(pbf, pbf.readVarint() + pbf.pos);
+    else if (tag === 11)
+      obj.description_text = new TranslatedStringReader().read(pbf, pbf.readVarint() + pbf.pos);
+  }
+}
+
+enum AlertCause {
+  UNKNOWN_CAUSE = 1,
+  OTHER_CAUSE,
+  TECHNICAL_PROBLEM,
+  STRIKE,
+  DEMONSTRATION,
+  ACCIDENT,
+  HOLIDAY,
+  WEATHER,
+  MAINTENANCE,
+  CONSTRUCTION,
+  POLICE_ACTIVITY,
+  MEDICAL_EMERGENCY,
+}
+
+enum AlertEffect {
+  NO_SERVICE = 1,
+  REDUCED_SERVICE,
+  SIGNIFICANT_DELAYS,
+  ADDITIONAL_SERVICE,
+  MODIFIED_SERVICE,
+  OTHER_EFFECT,
+  UNKNOWN_EFFECT,
+  STOP_MOVED,
+}
+
+export class TimeRangeReader extends ProtobufMessageReader<TimeRange> {
+  readonly defaultMessage: TimeRange = {
+    start: 0,
+    end: 0,
+  };
+
+  protected readField(tag: number, obj = {} as TimeRange, pbf: Pbf): void {
+    if (tag === 1) obj.start = pbf.readVarint();
```

Comment:
IMO: Maybe it's better to attach specific name for these number.

Reply:
It could've been like that if all of them were part of common context(say 1 mean int, 2 mean float) but here each `readField` call has used them for different members of the corresponding class.

Since those const wouldn't have been used by other files, I decided to make it smaller.

Reply:
It makes sense.